static/js/components.ts

```diff
@@ -13,7 +15,21 @@ import * as keydown_util from "./keydown_util";
     }).get();
 */
 
-export function toggle(opts) {
+export function toggle(opts: {
+    html_class?: string;
+    values: {label: string; label_html?: string; key: string}[];
+    callback?: (label: string, value: string) => void;
+    child_wants_focus?: boolean;
+    selected?: number;
+}): {
+    maybe_go_left: () => boolean;
+    maybe_go_right: () => boolean;
+    disable_tab(name: string): void;
+    enable_tab(name: string): void;
+    value(): string | undefined;
+    get(): JQuery;
+    goto(name: string): void;
+} {
```

Comment:
We’ll want to name and export this return type as `Toggle`, since it acts as a pseudo-class, and gets stored in state variables in `info_overlay.js`, `settings_toggle.js`, `subs.js`.

(At some point, it’d be good to convert it to a real class too.)

Reply:
Done.